spec/requests/generic_object_definitions_spec.rb

```diff
@@ -76,6 +76,81 @@
     end
   end
 
+  describe 'GET /api/generic_object_definitions/:id/generic_objects' do
+    it 'can return the generic objects of a definition with appropriate role' do
+      object = FactoryGirl.create(:generic_object, :name => 'bar', :generic_object_definition => object_def)
+      api_basic_authorize subcollection_action_identifier(:generic_object_definitions, :generic_objects, :read, :get)
+
+      get(api_generic_object_definition_generic_objects_url(nil, object_def))
+
+      expected = {
+        'name'      => 'generic_objects',
+        'count'     => 1,
+        'subcount'  => 1,
+        'resources' => [
+          { 'href' => api_generic_object_definition_generic_object_url(nil, object_def.id, object.id) }
+        ]
+      }
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body).to include(expected)
+    end
+
+    it 'does not return the generic objects without an appropriate role' do
+      api_basic_authorize
+
+      get(api_generic_object_definition_generic_objects_url(nil, object_def))
+
+      expect(response).to have_http_status(:forbidden)
+    end
+  end
+
+  describe 'GET /api/generic_object_definitions/:id/generic_objects/:id' do
+    let(:object) { FactoryGirl.create(:generic_object, :name => 'bar', :generic_object_definition => object_def) }
+    let(:vm) { FactoryGirl.create(:vm_amazon) }
+
+    before do
+      object_def.add_property_attribute('is_something', 'boolean')
+      object_def.add_property_association('services', 'Service')
+      object_def.add_property_association('vms', 'Vm')
+
+      object.is_something = true
+      object.save!
+
+      object.add_to_property_association('vms', [vm])
+    end
+
+    it 'returns the generic object with property attributes by default' do
+      api_basic_authorize subcollection_action_identifier(:generic_object_definitions, :generic_objects, :read, :get)
+
+      get(api_generic_object_definition_generic_object_url(nil, object_def.id, object.id))
+
+      expected = {
+        'href'                => api_generic_object_definition_generic_object_url(nil, object_def.id, object.id),
+        'id'                  => object.id.to_s,
+        'name'                => 'bar',
+        'property_attributes' => { 'is_something' => true }
+      }
+      expect(response.parsed_body).to include(expected)
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'will return the specified associations' do
+      api_basic_authorize subcollection_action_identifier(:generic_object_definitions, :generic_objects, :read, :get)
+
+      get(api_generic_object_definition_generic_object_url(nil, object_def.id, object.id), :params => {:associations => 'vms'})
+
+      expected = {
+        'href'                => api_generic_object_definition_generic_object_url(nil, object_def.id, object.id),
```

Comment:
same comment throughout, with these helpers just the object so ...(nil, object_def, object)